src/compiler/checker.ts

```diff
@@ -10702,8 +10779,24 @@ namespace ts {
                 }
 
                 let derived = getTargetSymbol(getPropertyOfObjectType(type, base.name));
-                if (derived) {
-                    let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+                let baseDeclarationFlags = getDeclarationFlagsFromSymbol(base);
+
+                Debug.assert(!!derived, "derived should point to something, even if it is the base class' declaration.");
+
+                if (derived === base) { 

```

Comment:
The assertion on line 10790 (in that commit) fails with class expressions (where there is no decl associated with the expression obtained from `getDeclarationOfKind()`). The test for the error has been changed to reflect this. 

Is this what you meant by this issue?